File: week1/ShuffleArray/shuffleArray.py
import random
class Solution:
    def __init__(self, nums: list[int]):
        self.org = nums
        self.numbers = nums

    def shuffle(self):
        sf = list(self.numbers)
        # m = len(sf)
        # for i in range(m):
        #     tmp = random.randrange(i, m)
        #     sf[i], sf[tmp] =  sf[tmp], sf[i]
        # or just use the shuffle func --- 
        random.shuffle(sf)
        self.numbers = sf

    def reset(self):
        self.numbers = self.org
    
input = [1,2,3]
output = Solution(input)
print(output.numbers)
output.shuffle()
print(output.numbers)
output.reset()
print(output.numbers)

File: week1/ShuffleArray/shuffleArray.cpp
#include <iostream>
using namespace std;

class shuffleArray{
    vector<int> org;
    int n;
public:
    void set(vector<int>numbers){
        this->org = numbers;
        this->n = numbers.size();
        cout << "this is org size list: " << n << "\n";
    }

    vector<int> reset(){
        return org;
    }

    vector<int> shuffle(){
        vector<int> shuffle = org;
        int randNum = n;
        for(int i=0; i< n; i++){
            int tmp = rand()%randNum;
            swap(shuffle[i], shuffle[tmp]);
        }

        return shuffle;
    }
};

int main(){
    vector<int> input = {1,2,3};
    shuffleArray arr;
    arr.set(input);
    vector<int> output = arr.shuffle();
    for(auto it: output){
        cout << it << " ";
    }
    vector<int> outputReset = arr.reset();
    for(auto it: outputReset){
        cout << it << " ";
    }
}